pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -472,49 +474,49 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
 // GetProxyServiceInstances returns service instances co-located with a given proxy
 func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.ServiceInstance, error) {
 	proxyIP := proxy.IPAddress
-	podNamespace := ""
-	if pod, exists := c.pods.getPodByIP(proxyIP); exists {
-		podNamespace = pod.Namespace
+	pod, exists := c.pods.getPodByIP(proxyIP)
+	if !exists {
+		return nil, fmt.Errorf("failed to get pod by proxyIP %s", proxyIP)
```

Comment:
This is very likely to break a lot of stuff - from multicluster to mesh expansion to who knows what. 

PLEASE REVERT !

It is called from aggregate, and the IP of the node doesn't have to be in each k8s registry - or in any at all ( consul, etc)

Reply:
Do we have any tests to catch this?

Reply:
I am so sorry for missing background knowledge, will propose a fix ASAP.

Reply:
the `podNamespace` variable is really misleading, I read it and thought the proxy in k8s must be an existing pod.

Reply:
I know. Maybe call `proxyNamespace` or whaterver.

Reply:
Yes, in the old code if a registry doesn't find anything - it returns nothing, and lets other registry that may know about the IP return. 

I wonder why the multi-cluster tests didn't catch this - I suspect the tests are 'lucky' and the test are covering pods in the 'first' registry.